{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [],
   "source": [
    "from __future__ import (division, print_function, absolute_import,unicode_literals)\n",
    "import sys,os,shutil\n",
    "import subprocess\n",
    "#import simplejson\n",
    "import logging \n",
    "import time\n",
    "import operator\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [],
   "source": [
    "def log(message):\n",
    "    \"\"\" Log messages to standard output. \"\"\"\n",
    "    print(time.ctime() + ' --- ' + message, flush=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [],
   "source": [
    "def run_cmd(cmnd):\n",
    "  try:\n",
    "     output = subprocess.check_output(\n",
    "        cmnd, stderr=subprocess.STDOUT, shell=True,\n",
    "        universal_newlines=True)\n",
    "  except subprocess.CalledProcessError as exc:\n",
    "     print(\"Status : FAIL\", exc.returncode, exc.output)\n",
    "  else:\n",
    "    print(\"Output: \\n{}\\n\".format(output))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "env: PATH=$PATH:/home/idies/workspace/Storage/sramakr4/persistent/bin:/home/idies/workspace/Storage/sramakr4/persistent/miniconda3/envs/nextstrain/bin:/home/idies/miniconda3/condabin:/home/idies/.local/bin:/home/idies/bin:/home/idies/R/bin:/home/idies/miniconda3/bin:/usr/local/texlive/bin/x86_64-linux:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin\n"
     ]
    }
   ],
   "source": [
    "# Setup environment to run nextstrain\n",
    "#run_cmd(\"/home/idies/workspace/Storage/sramakr4/persistent/bashrc\")\n",
    "%env PATH=$PATH:/home/idies/workspace/Storage/sramakr4/persistent/bin:/home/idies/workspace/Storage/sramakr4/persistent/miniconda3/envs/nextstrain/bin:/home/idies/miniconda3/condabin:/home/idies/.local/bin:/home/idies/bin:/home/idies/R/bin:/home/idies/miniconda3/bin:/usr/local/texlive/bin/x86_64-linux:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin\n",
    "                                  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [],
   "source": [
    "NEXTSTRAIN_BUILD_DIR=\"/home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [],
   "source": [
    "#run_cmd(\"mkdir /home/idies/workspace/covid19/nextstrain/archive\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [],
   "source": [
    "out_path = '/home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/final/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [],
   "source": [
    "#run_cmd(\"/home/idies/workspace/Storage/sramakr4/persistent/miniconda3/envs/nextstrain/bin/nextstrain --help\")\n",
    "#run_cmd(\"snakemake --help \")\n",
    "#run_cmd(\"curl -o \" + out_path + \"gisaid_metadata.tsv\" + \" \" + \"https://github.com/nextstrain/ncov/blob/master/data/metadata.tsv\" )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status : FAIL 1 \n",
      "Usage:   seqtk <command> <arguments>\n",
      "Version: 1.3-r106\n",
      "\n",
      "Command: seq       common transformation of FASTA/Q\n",
      "         comp      get the nucleotide composition of FASTA/Q\n",
      "         sample    subsample sequences\n",
      "         subseq    extract subsequences from FASTA/Q\n",
      "         fqchk     fastq QC (base/quality summary)\n",
      "         mergepe   interleave two PE FASTA/Q files\n",
      "         trimfq    trim FASTQ using the Phred algorithm\n",
      "\n",
      "         hety      regional heterozygosity\n",
      "         gc        identify high- or low-GC regions\n",
      "         mutfa     point mutate FASTA at specified positions\n",
      "         mergefa   merge two FASTA/Q files\n",
      "         famask    apply a X-coded FASTA to a source FASTA\n",
      "         dropse    drop unpaired from interleaved PE FASTA/Q\n",
      "         rename    rename sequence names\n",
      "         randbase  choose a random base from hets\n",
      "         cutN      cut sequence at long N\n",
      "         listhet   extract the position of each het\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#run_cmd(\"nextstrain --help \")\n",
    "#run_cmd(\"snakemake --help \")\n",
    "run_cmd(\"seqtk\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output: \n",
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 4\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job counts:\n",
      "\tcount\tjobs\n",
      "\t1\tall_data\n",
      "\t1\tdownload_gisaid\n",
      "\t1\tmerge_seq\n",
      "\t1\tnormalize_gisaid\n",
      "\t1\tnormalize_local\n",
      "\t1\trun_only_jhu_seq\n",
      "\t6\n",
      "\n",
      "[Tue Jun  9 01:29:15 2020]\n",
      "Job 4: \n",
      "        Parse local metadata to nextstrain format and Prepare JHU sequences\n",
      "        \n",
      "\n",
      "\n",
      "[Tue Jun  9 01:29:15 2020]\n",
      "Job 5: Downloading metadata and fasta files from GISAID\n",
      "\n",
      "Add_metadata : yes\n",
      "Running Cmd : scripts/parse_meta_from_GISAID_to_nextstrain.py -in /home/idies/workspace/covid19/jhu_sequences/submissions/all_submitted_jhu_sequences_metadata.tsv -g /home/idies/workspace/covid19/jhu_sequences/submissions/all_submitted_jhu_sequences.fasta -m yes -o /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/final/JHU_nextstrain.tsv\n",
      "2020-06-08\n",
      "Total Number of Samples : 115\n",
      "Total Number of Samples  : 115\n",
      "['strain', 'virus', 'gisaid_epi_isl', 'genbank_accession', 'date', 'region', 'country', 'division', 'location', 'region_exposure', 'country_exposure', 'division_exposure', 'segment', 'length', 'host', 'age', 'sex', 'originating_lab', 'submitting_lab', 'authors', 'url', 'title', 'date_submitted', 'run_date', 'run_id', 'race', 'days_from_onset', 'diagnostic_ct_value', 'pangolin_clade', 'travel', 'covidsickcontact', 'cardiacdx', 'lungdx', 'diabetes', 'obese', 'etoh', 'smoker', 'immunecomp', 'fever', 'bodyache', 'gi', 'sob', 'icu', 'vent', 'oxygen', 'nsaid', 'anyadmission', 'cxrabn', 'outcome', '_us_travel_location1', 'us_travel_location3', 'international_travel_location_1', 'international_travel_location_2', 'international_travel_location_3', 'cough']\n",
      "['strain', 'date', 'host', 'sex', 'age', 'originating_lab', 'submitting_lab', 'authors', 'date_submitted', 'run_date', 'run_id', 'race', 'days_from_onset', 'diagnostic_ct_value', 'pangolin_clade', 'travel', 'covidsickcontact', 'cardiacdx', 'lungdx', 'diabetes', 'obese', 'etoh', 'smoker', 'immunecomp', 'fever', 'bodyache', 'gi', 'sob', 'icu', 'vent', 'oxygen', 'nsaid', 'anyadmission', 'cxrabn', 'outcome', '_us_travel_location1', 'us_travel_location3', 'international_travel_location_1', 'international_travel_location_2', 'international_travel_location_3', 'cough', 'region', 'country', 'division', 'location', 'gisaid_epi_isl', 'virus', 'genbank_accession', 'region_exposure', 'country_exposure', 'division_exposure', 'segment', 'length', 'url', 'title']\n",
      "Processed Samples : 115 ; Metadata fields : 55\n",
      "2020-06-08\n",
      "Total Number of Samples : 115\n",
      "Total Number of Samples  : 115\n",
      "['strain', 'virus', 'gisaid_epi_isl', 'genbank_accession', 'date', 'region', 'country', 'division', 'location', 'region_exposure', 'country_exposure', 'division_exposure', 'segment', 'length', 'host', 'age', 'sex', 'originating_lab', 'submitting_lab', 'authors', 'url', 'title', 'date_submitted', 'run_date', 'run_id', 'race', 'days_from_onset', 'diagnostic_ct_value', 'pangolin_clade', 'travel', 'covidsickcontact', 'cardiacdx', 'lungdx', 'diabetes', 'obese', 'etoh', 'smoker', 'immunecomp', 'fever', 'bodyache', 'gi', 'sob', 'icu', 'vent', 'oxygen', 'nsaid', 'anyadmission', 'cxrabn', 'outcome', '_us_travel_location1', 'us_travel_location3', 'international_travel_location_1', 'international_travel_location_2', 'international_travel_location_3', 'cough']\n",
      "['strain', 'date', 'host', 'sex', 'age', 'originating_lab', 'submitting_lab', 'authors', 'date_submitted', 'run_date', 'run_id', 'race', 'days_from_onset', 'diagnostic_ct_value', 'pangolin_clade', 'travel', 'covidsickcontact', 'cardiacdx', 'lungdx', 'diabetes', 'obese', 'etoh', 'smoker', 'immunecomp', 'fever', 'bodyache', 'gi', 'sob', 'icu', 'vent', 'oxygen', 'nsaid', 'anyadmission', 'cxrabn', 'outcome', '_us_travel_location1', 'us_travel_location3', 'international_travel_location_1', 'international_travel_location_2', 'international_travel_location_3', 'cough', 'region', 'country', 'division', 'location', 'gisaid_epi_isl', 'virus', 'genbank_accession', 'region_exposure', 'country_exposure', 'division_exposure', 'segment', 'length', 'url', 'title']\n",
      "Processed Samples : 115 ; Metadata fields : 55\n",
      "Running Cmd: scripts/get_add_meta_columns.sh /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/final/JHU_nextstrain.tsv 23 /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/final/add_metadata_list.txt\n",
      "Running Cmd : scripts/normalize_and_filter_local_fasta.sh /home/idies/workspace/covid19/jhu_sequences/submissions/all_submitted_jhu_sequences.fasta /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/final/JHU_nextstrain.tsv /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/final/JHU_sequences.fasta /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/final/JHU_metadata.tsv 29902\n",
      "Normalizing GISAID file /home/idies/workspace/covid19/jhu_sequences/submissions/all_submitted_jhu_sequences.fasta  (min length 29902)\n",
      "Filtering GISAID file /home/idies/workspace/covid19/jhu_sequences/submissions/all_submitted_jhu_sequences.fasta based on metadata\n",
      "[Tue Jun  9 01:29:31 2020]\n",
      "Finished job 5.\n",
      "1 of 6 steps (17%) done\n",
      "[Tue Jun  9 01:29:33 2020]\n",
      "Finished job 4.\n",
      "2 of 6 steps (33%) done\n",
      "\n",
      "[Tue Jun  9 01:29:33 2020]\n",
      "Job 3: \n",
      "        Normalize GISAID sequences and metadata\n",
      "        \n",
      "\n",
      "Running Cmd : scripts/normalize_and_filter_gisaid_fasta.sh /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/final/gisaid_raw.fasta /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/final/gisaid_raw_meta.tsv /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/final/gisaid_sequences.fasta /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/final/gisaid_metadata.tsv\n",
      "Using default minimum length of 25000\n",
      "Normalizing GISAID file /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/final/gisaid_raw.fasta  (min length 29002)\n",
      "Filtering GISAID file /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/final/gisaid_raw.fasta based on metadata\n",
      "Adding additional metadata run_data and run_id\n",
      "['virus', 'gisaid_epi_isl', 'genbank_accession', 'date', 'region', 'country', 'division', 'location', 'region_exposure', 'country_exposure', 'division_exposure', 'segment', 'length', 'host', 'age', 'sex', 'originating_lab', 'submitting_lab', 'authors', 'url', 'title', 'paper_url', 'date_submitted']\n",
      "['virus', 'gisaid_epi_isl', 'genbank_accession', 'date', 'region', 'country', 'division', 'location', 'region_exposure', 'country_exposure', 'division_exposure', 'segment', 'length', 'host', 'age', 'sex', 'originating_lab', 'submitting_lab', 'authors', 'url', 'title', 'date_submitted', 'run_date', 'run_id', 'race', 'days_from_onset', 'diagnostic_ct_value', 'pangolin_clade', 'travel', 'covidsickcontact', 'cardiacdx', 'lungdx', 'diabetes', 'obese', 'etoh', 'smoker', 'immunecomp', 'fever', 'bodyache', 'gi', 'sob', 'icu', 'vent', 'oxygen', 'nsaid', 'anyadmission', 'cxrabn', 'outcome', '_us_travel_location1', 'us_travel_location3', 'international_travel_location_1', 'international_travel_location_2', 'international_travel_location_3', 'cough']\n",
      "(36297, 23)\n",
      "Index(['Algeria/G0638_2264/2020', 'Algeria/G0640_2265/2020',\n",
      "       'Algeria/G0860_2262/2020', 'Anhui/SZ005/2020', 'Argentina/C121/2020',\n",
      "       'Argentina/C1374/2020', 'Argentina/C3013/2020',\n",
      "       'Argentina/PAIS_A001/2020', 'Argentina/PAIS_A003/2020',\n",
      "       'Argentina/PAIS_A004/2020',\n",
      "       ...\n",
      "       'mink/Netherlands/NB03_index/2020', 'mink/Netherlands/NB04_index/2020',\n",
      "       'mouse/Harbin/HRB-26m/2020', 'pangolin/Guangdong/1/2019',\n",
      "       'pangolin/Guangxi/P1E/2017', 'pangolin/Guangxi/P2V/2017',\n",
      "       'pangolin/Guangxi/P4L/2017', 'pangolin/Guangxi/P5E/2017',\n",
      "       'pangolin/Guangxi/P5L/2017', 'tiger/USA/NY-040420/2020'],\n",
      "      dtype='object', name='strain', length=36297)\n",
      "(115, 54)\n",
      "Index(['USA/DC-HP00006/2020', 'USA/DC-HP00007/2020', 'USA/MD-HP00010/2020',\n",
      "       'USA/MD-HP00015/2020', 'USA/MD-HP00016/2020', 'USA/MD-HP00018/2020',\n",
      "       'USA/MD-HP00020/2020', 'USA/DC-HP00022/2020', 'USA/MD-HP00023/2020',\n",
      "       'USA/MD-HP00029/2020',\n",
      "       ...\n",
      "       'USA/DC-HP00122/2020', 'USA/MD-HP00123/2020', 'USA/MD-HP00127/2020',\n",
      "       'USA/MD-HP00128/2020', 'USA/MD-HP00129/2020', 'USA/MD-HP00131/2020',\n",
      "       'USA/MD-HP00132/2020', 'USA/MD-HP00133/2020', 'USA/DC-HP00135/2020',\n",
      "       'USA/DC-HP00137/2020'],\n",
      "      dtype='object', name='strain', length=115)\n",
      "(36297, 55)\n",
      "<bound method NDFrame.head of                           virus  ...    cough\n",
      "strain                           ...         \n",
      "Algeria/G0638_2264/2020    ncov  ...  unknown\n",
      "Algeria/G0640_2265/2020    ncov  ...  unknown\n",
      "Algeria/G0860_2262/2020    ncov  ...  unknown\n",
      "Anhui/SZ005/2020           ncov  ...  unknown\n",
      "Argentina/C121/2020        ncov  ...  unknown\n",
      "...                         ...  ...      ...\n",
      "pangolin/Guangxi/P2V/2017  ncov  ...  unknown\n",
      "pangolin/Guangxi/P4L/2017  ncov  ...  unknown\n",
      "pangolin/Guangxi/P5E/2017  ncov  ...  unknown\n",
      "pangolin/Guangxi/P5L/2017  ncov  ...  unknown\n",
      "tiger/USA/NY-040420/2020   ncov  ...  unknown\n",
      "\n",
      "[36297 rows x 55 columns]>\n",
      "INFO : Added additional metadata to existing metadata file\n",
      "[Tue Jun  9 01:31:24 2020]\n",
      "Finished job 3.\n",
      "3 of 6 steps (50%) done\n",
      "\n",
      "[Tue Jun  9 01:31:24 2020]\n",
      "Job 2: \n",
      "        Merge JHU sequences and reference.fasta\n",
      "        \n",
      "\n",
      "\n",
      "[Tue Jun  9 01:31:25 2020]\n",
      "Job 1: \n",
      "        Merge JHU sequences and local sequences\n",
      "        \n",
      "\n",
      "Running Cmd : scripts/get_ref_sequences.sh /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/final/gisaid_sequences.fasta /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/final/gisaid_metadata.tsv config/include.txt /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/final/only_ref_sequences.fasta /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/final/only_ref_metadata.tsv\n",
      "Getting the reference only seq and metadata from REF file /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/final/gisaid_sequences.fasta\n",
      "Running Cmd : scripts/combine_seq.sh /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/final/gisaid_sequences.fasta /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/final/gisaid_metadata.tsv /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/final/JHU_sequences.fasta /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/final/JHU_metadata.tsv /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/final/sequences.fasta /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/final/metadata.tsv\n",
      "\n",
      "Combining Input sequences for nextstrain GISAID\n",
      "Filtering REF file /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/final/gisaid_sequences.fasta based on metadata\n",
      " Running Cmd  : scripts/combine_seq.sh /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/final/only_ref_sequences.fasta /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/final/only_ref_metadata.tsv /home/idies/workspace/covid19/jhu_sequences/submissions/all_submitted_jhu_sequences.fasta /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/final/all_JHU_nextstrain.tsv /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/final/local_sequences.fasta /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/final/local_metadata.tsv\n",
      "Combining Input sequences for nextstrain GISAID\n",
      "[Tue Jun  9 01:31:44 2020]\n",
      "Finished job 2.\n",
      "4 of 6 steps (67%) done\n",
      "[Tue Jun  9 01:31:45 2020]\n",
      "Finished job 1.\n",
      "5 of 6 steps (83%) done\n",
      "\n",
      "[Tue Jun  9 01:31:45 2020]\n",
      "localrule all_data:\n",
      "    input: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/final/sequences.fasta, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/final/metadata.tsv, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/final/local_sequences.fasta, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/final/local_metadata.tsv\n",
      "    jobid: 0\n",
      "\n",
      "[Tue Jun  9 01:31:45 2020]\n",
      "Finished job 0.\n",
      "6 of 6 steps (100%) done\n",
      "Complete log: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/.snakemake/log/2020-06-09T012914.561854.snakemake.log\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Execute nextstrain from this path:\n",
    "os.chdir(NEXTSTRAIN_BUILD_DIR)\n",
    "\n",
    "#run_cmd(\"snakemake -j 4 --forceall --snakefile Snakefile_prepare\")\n",
    "#run_cmd(\"snakemake --unlock\")\n",
    "#run_cmd(\"snakemake -j 4 --forceall --snakefile rules/prepare.smk\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Output: \n",
      "Building DAG of jobs...\n",
      "region_global\n",
      "OrderedDict([('global', OrderedDict([('group_by', 'division year month'), ('seq_per_group', 10), ('include', \"--include-where 'originating_lab=Johns Hopkins Hospital Department of Pathology'\")]))])\n",
      "global\n",
      "['global']\n",
      "region_global\n",
      "region\n",
      "OrderedDict([('region', OrderedDict([('group_by', 'division year month'), ('seq_per_group', 96), ('exclude', \"--exclude-where 'region!={region}'\"), ('include', \"--include-where 'originating_lab=Johns Hopkins Hospital Department of Pathology'\")])), ('global', OrderedDict([('group_by', 'country year month'), ('seq_per_group', 10), ('exclude', \"--exclude-where 'region={region}'\"), ('include', \"--include-where 'originating_lab=Johns Hopkins Hospital Department of Pathology'\"), ('priorities', OrderedDict([('type', 'proximity'), ('focus', 'region')]))]))])\n",
      "north-america\n",
      "['region', 'global']\n",
      "region\n",
      "region\n",
      "Job counts:\n",
      "\tcount\tjobs\n",
      "\t1\tadjust_metadata_regions\n",
      "\t1\taggregate_alignments\n",
      "\t1\tall\n",
      "\t2\tancestral\n",
      "\t2\tclades\n",
      "\t2\tcolors\n",
      "\t2\tcombine_samples\n",
      "\t2\tdated_json\n",
      "\t1\tdiagnostic\n",
      "\t1\tdownload\n",
      "\t2\texport\n",
      "\t1\tfilter\n",
      "\t2\tfinalize\n",
      "\t2\tincorporate_travel_history\n",
      "\t2\tlegacy_clades\n",
      "\t1\tmask\n",
      "\t2\tpangolin\n",
      "\t1\tpartition_sequences\n",
      "\t1\tproximity_score\n",
      "\t2\trecency\n",
      "\t1\trefilter\n",
      "\t2\trefine\n",
      "\t2\trename_legacy_clades\n",
      "\t3\tsubsample\n",
      "\t2\ttip_frequencies\n",
      "\t2\ttraits\n",
      "\t2\ttranslate\n",
      "\t2\ttree\n",
      "\t47\n",
      "\n",
      "[Tue Jun  9 01:47:47 2020]\n",
      "Job 12: Downloading metadata and fasta files from S3\n",
      "Reason: Missing output files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/data/metadata.tsv, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/data/sequences.fasta; Updated input files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/final/sequences.fasta, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/final/metadata.tsv\n",
      "\n",
      "\n",
      "        cp /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/final/sequences.fasta /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/data/sequences.fasta\n",
      "        cp /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/final/metadata.tsv /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/data/metadata.tsv\n",
      "        #aws s3 cp s3://nextstrain-ncov-private/metadata.tsv.gz - | gunzip -cq >/home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/data/metadata.tsv\n",
      "        #aws s3 cp s3://nextstrain-ncov-private/sequences.fasta.gz - | gunzip -cq > /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/data/sequences.fasta\n",
      "        \n",
      "\n",
      "[Tue Jun  9 01:47:47 2020]\n",
      "Job 10: Constructing colors file\n",
      "Reason: Missing output files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/colors.tsv; Input files updated by another job: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/data/metadata.tsv\n",
      "\n",
      "\n",
      "        /home/idies/workspace/Storage/sramakr4/persistent/miniconda3/envs/nextstrain/bin/python scripts/assign-colors.py             --ordering config/ordering.tsv             --color-schemes config/color_schemes.tsv             --output /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/colors.tsv             --metadata /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/data/metadata.tsv 2>&1 | tee /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/logs/colors_global.txt\n",
      "        \n",
      "\n",
      "[Tue Jun  9 01:47:47 2020]\n",
      "Job 50: \n",
      "        Filtering to\n",
      "          - excluding strains in config/exclude.txt\n",
      "        \n",
      "Reason: Missing output files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/filtered.fasta; Input files updated by another job: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/data/metadata.tsv, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/data/sequences.fasta\n",
      "\n",
      "\n",
      "        augur filter             --sequences /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/data/sequences.fasta             --metadata /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/data/metadata.tsv             --include config/include.txt             --exclude config/exclude.txt             --exclude-where division='USA' date='2020' date='2020-01-XX' date='2020-02-XX' date='2020-03-XX' date='2020-04-XX' date='2020-05-XX' date='2020-06-XX' date='2020-01' date='2020-02' date='2020-03' date='2020-04' date='2020-05' date='2020-06'            --include-where originating_lab='Johns Hopkins Hospital Department of Pathology'            --min-length 27000             --output /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/filtered.fasta 2>&1 | tee /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/logs/filtered.txt\n",
      "        \n",
      "\n",
      "[Tue Jun  9 01:47:47 2020]\n",
      "Job 22: Use metadata on submission date to construct submission recency field\n",
      "Reason: Missing output files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/recency.json; Input files updated by another job: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/data/metadata.tsv\n",
      "\n",
      "\n",
      "        /home/idies/workspace/Storage/sramakr4/persistent/miniconda3/envs/nextstrain/bin/python scripts/construct-recency-from-submission-date.py             --metadata /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/data/metadata.tsv             --output /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/recency.json 2>&1 | tee /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/logs/recency_global.txt\n",
      "        \n",
      "\n",
      "[Tue Jun  9 01:47:48 2020]\n",
      "Job 16: \n",
      "        Adjusting metadata for build 'north-america'\n",
      "        \n",
      "Reason: Missing output files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/metadata_adjusted.tsv; Input files updated by another job: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/data/metadata.tsv\n",
      "\n",
      "\n",
      "        /home/idies/workspace/Storage/sramakr4/persistent/miniconda3/envs/nextstrain/bin/python scripts/adjust_regional_meta.py             --region 'North America'             --metadata /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/data/metadata.tsv             --output /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/metadata_adjusted.tsv 2>&1 | tee /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/logs/adjust_metadata_regions_north-america.txt\n",
      "        \n",
      "\n",
      "[Tue Jun  9 01:47:48 2020]\n",
      "checkpoint partition_sequences:\n",
      "    input: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/filtered.fasta\n",
      "    output: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/split_sequences/\n",
      "    log: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/logs/partition_sequences.txt\n",
      "    jobid: 49\n",
      "    reason: Missing output files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/split_sequences/; Input files updated by another job: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/filtered.fasta\n",
      "Downstream jobs will be updated after completion.\n",
      "\n",
      "\n",
      "        /home/idies/workspace/Storage/sramakr4/persistent/miniconda3/envs/nextstrain/bin/python scripts/partition-sequences.py             --sequences /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/filtered.fasta             --sequences-per-group 150             --output-dir /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/split_sequences/ 2>&1 | tee /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/logs/partition_sequences.txt\n",
      "        \n",
      "\n",
      "[Tue Jun  9 01:47:48 2020]\n",
      "Job 32: Use metadata on submission date to construct submission recency field\n",
      "Reason: Missing output files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/recency.json; Input files updated by another job: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/metadata_adjusted.tsv\n",
      "\n",
      "\n",
      "        /home/idies/workspace/Storage/sramakr4/persistent/miniconda3/envs/nextstrain/bin/python scripts/construct-recency-from-submission-date.py             --metadata /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/metadata_adjusted.tsv             --output /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/recency.json 2>&1 | tee /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/logs/recency_north-america.txt\n",
      "        \n",
      "\n",
      "[Tue Jun  9 01:47:48 2020]\n",
      "Job 14: Constructing colors file\n",
      "Reason: Missing output files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/colors.tsv; Input files updated by another job: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/metadata_adjusted.tsv\n",
      "\n",
      "\n",
      "        /home/idies/workspace/Storage/sramakr4/persistent/miniconda3/envs/nextstrain/bin/python scripts/assign-colors.py             --ordering config/ordering.tsv             --color-schemes config/color_schemes.tsv             --output /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/colors.tsv             --metadata /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/metadata_adjusted.tsv 2>&1 | tee /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/logs/colors_north-america.txt\n",
      "        \n",
      "\n",
      "[Tue Jun  9 01:47:48 2020]\n",
      "Job 47: Collecting alignments\n",
      "Reason: Missing output files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/aligned.fasta; Input files updated by another job: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/split_sequences/\n",
      "\n",
      "\n",
      "        cat /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/split_sequences/ > /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/aligned.fasta 2> /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/logs/aggregate_alignments.txt\n",
      "        \n",
      "\n",
      "[Tue Jun  9 01:47:48 2020]\n",
      "Job 48: Scanning aligned sequences /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/aligned.fasta for problematic sequences\n",
      "Reason: Missing output files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/to-exclude.txt; Input files updated by another job: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/data/metadata.tsv, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/aligned.fasta\n",
      "\n",
      "\n",
      "        /home/idies/workspace/Storage/sramakr4/persistent/miniconda3/envs/nextstrain/bin/python scripts/diagnostic.py             --alignment /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/aligned.fasta             --metadata /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/data/metadata.tsv             --reference config/reference.gb             --mask-from-beginning 100             --mask-from-end 50             --output-flagged /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/flagged-sequences.tsv             --output-diagnostics /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/sequence-diagnostics.tsv             --output-exclusion-list /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/to-exclude.txt 2>&1 | tee /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/logs/diagnostics.txt\n",
      "        \n",
      "\n",
      "[Tue Jun  9 01:47:48 2020]\n",
      "Job 46: \n",
      "        excluding sequences flagged in the diagnostic step in file /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/to-exclude.txt\n",
      "        \n",
      "Reason: Missing output files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/aligned-filtered.fasta; Input files updated by another job: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/to-exclude.txt, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/data/metadata.tsv, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/aligned.fasta\n",
      "\n",
      "\n",
      "        augur filter             --sequences /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/aligned.fasta             --metadata /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/data/metadata.tsv             --exclude /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/to-exclude.txt             --output /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/aligned-filtered.fasta 2>&1 | tee /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/logs/refiltered.txt\n",
      "        \n",
      "\n",
      "[Tue Jun  9 01:47:48 2020]\n",
      "Job 43: \n",
      "        Mask bases in alignment\n",
      "          - masking 100 from beginning\n",
      "          - masking 50 from end\n",
      "          - masking other sites: 13402 24389 24390\n",
      "        \n",
      "Reason: Missing output files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/masked.fasta; Input files updated by another job: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/aligned-filtered.fasta\n",
      "\n",
      "\n",
      "        /home/idies/workspace/Storage/sramakr4/persistent/miniconda3/envs/nextstrain/bin/python scripts/mask-alignment.py             --alignment /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/aligned-filtered.fasta             --mask-from-beginning 100             --mask-from-end 50             --mask-sites 13402 24389 24390             --mask-terminal-gaps             --output /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/masked.fasta 2>&1 | tee /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/logs/mask.txt\n",
      "        \n",
      "region\n",
      "region\n",
      "region\n",
      "region\n",
      "region\n",
      "\n",
      "[Tue Jun  9 01:47:48 2020]\n",
      "Job 41: \n",
      "        Subsample all sequences into a region set for build 'north-america' with 96 per division year month\n",
      "        \n",
      "Reason: Missing output files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/sample-region.fasta; Input files updated by another job: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/masked.fasta, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/data/metadata.tsv\n",
      "\n",
      "\n",
      "        augur filter             --sequences /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/masked.fasta             --metadata /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/data/metadata.tsv             --include config/include.txt             --exclude-where 'region!=North America'             --include-where 'originating_lab=Johns Hopkins Hospital Department of Pathology'                          --group-by division year month             --sequences-per-group 96             --output /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/sample-region.fasta 2>&1 | tee \n",
      "        \n",
      "region_global\n",
      "region_global\n",
      "region_global\n",
      "region_global\n",
      "region_global\n",
      "\n",
      "[Tue Jun  9 01:47:48 2020]\n",
      "Job 37: \n",
      "        Subsample all sequences into a global set for build 'global' with 10 per division year month\n",
      "        \n",
      "Reason: Missing output files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/sample-global.fasta; Input files updated by another job: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/masked.fasta, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/data/metadata.tsv\n",
      "\n",
      "\n",
      "        augur filter             --sequences /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/masked.fasta             --metadata /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/data/metadata.tsv             --include config/include.txt                          --include-where 'originating_lab=Johns Hopkins Hospital Department of Pathology'                          --group-by division year month             --sequences-per-group 10             --output /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/sample-global.fasta 2>&1 | tee \n",
      "        \n",
      "\n",
      "[Tue Jun  9 01:47:48 2020]\n",
      "Job 25: \n",
      "        Combine and deduplicate FASTAs\n",
      "        \n",
      "Reason: Missing output files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/subsampled_alignment.fasta; Input files updated by another job: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/sample-global.fasta\n",
      "\n",
      "\n",
      "        /home/idies/workspace/Storage/sramakr4/persistent/miniconda3/envs/nextstrain/bin/python scripts/combine-and-dedup-fastas.py             --input /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/sample-global.fasta             --output /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/subsampled_alignment.fasta 2>&1 | tee /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/logs/subsample_regions_global.txt\n",
      "        \n",
      "\n",
      "[Tue Jun  9 01:47:48 2020]\n",
      "Job 45: \n",
      "        determine priority for inclusion in as phylogenetic context by\n",
      "        genetic similiarity to sequences in focal set for build 'north-america'.\n",
      "        \n",
      "Reason: Missing output files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/proximity_region.tsv; Input files updated by another job: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/masked.fasta, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/data/metadata.tsv, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/sample-region.fasta\n",
      "\n",
      "\n",
      "        /home/idies/workspace/Storage/sramakr4/persistent/miniconda3/envs/nextstrain/bin/python scripts/priorities.py --alignment /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/masked.fasta             --metadata /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/data/metadata.tsv             --reference config/reference.gb             --focal-alignment /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/sample-region.fasta             --output /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/proximity_region.tsv 2>&1 | tee /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/logs/subsampling_priorities_north-america_region.txt\n",
      "        \n",
      "region\n",
      "region\n",
      "region\n",
      "region\n",
      "region\n",
      "region\n",
      "\n",
      "[Tue Jun  9 01:47:48 2020]\n",
      "Job 42: \n",
      "        Subsample all sequences into a global set for build 'north-america' with 10 per country year month\n",
      "        \n",
      "Reason: Missing output files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/sample-global.fasta; Input files updated by another job: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/masked.fasta, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/proximity_region.tsv, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/data/metadata.tsv\n",
      "\n",
      "\n",
      "        augur filter             --sequences /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/masked.fasta             --metadata /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/data/metadata.tsv             --include config/include.txt             --exclude-where 'region=North America'             --include-where 'originating_lab=Johns Hopkins Hospital Department of Pathology'             --priority /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/proximity_region.tsv             --group-by country year month             --sequences-per-group 10             --output /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/sample-global.fasta 2>&1 | tee \n",
      "        \n",
      "\n",
      "[Tue Jun  9 01:47:48 2020]\n",
      "Job 24: Building tree\n",
      "Reason: Missing output files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/tree_raw.nwk; Input files updated by another job: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/subsampled_alignment.fasta\n",
      "\n",
      "\n",
      "        augur tree             --alignment /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/subsampled_alignment.fasta             --tree-builder-args '-ninit 10 -n 4'             --output /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/tree_raw.nwk             --nthreads 4 2>&1 | tee /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/logs/tree_global.txt\n",
      "        \n",
      "\n",
      "[Tue Jun  9 01:47:48 2020]\n",
      "Job 11: \n",
      "        Refining tree\n",
      "          - estimate timetree\n",
      "          - use skyline coalescent timescale\n",
      "          - estimate marginal node dates\n",
      "        \n",
      "Reason: Missing output files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/branch_lengths.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/tree.nwk; Input files updated by another job: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/subsampled_alignment.fasta, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/data/metadata.tsv, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/tree_raw.nwk\n",
      "\n",
      "\n",
      "        augur refine             --tree /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/tree_raw.nwk             --alignment /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/subsampled_alignment.fasta             --metadata /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/data/metadata.tsv             --output-tree /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/tree.nwk             --output-node-data /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/branch_lengths.json             --root Wuhan-Hu-1/2019 Wuhan/WH01/2019             --timetree             --clock-rate 0.0008             --clock-std-dev 0.0004             --coalescent skyline             --date-inference marginal             --divergence-unit mutations             --date-confidence             --no-covariance             --clock-filter-iqd 4 2>&1 | tee /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/logs/refine_global.txt\n",
      "        \n",
      "\n",
      "[Tue Jun  9 01:47:48 2020]\n",
      "Job 35: \n",
      "        Combine and deduplicate FASTAs\n",
      "        \n",
      "Reason: Missing output files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/subsampled_alignment.fasta; Input files updated by another job: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/sample-global.fasta, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/sample-region.fasta\n",
      "\n",
      "\n",
      "        /home/idies/workspace/Storage/sramakr4/persistent/miniconda3/envs/nextstrain/bin/python scripts/combine-and-dedup-fastas.py             --input /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/sample-region.fasta /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/sample-global.fasta             --output /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/subsampled_alignment.fasta 2>&1 | tee /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/logs/subsample_regions_north-america.txt\n",
      "        \n",
      "\n",
      "[Tue Jun  9 01:47:48 2020]\n",
      "Job 34: Building tree\n",
      "Reason: Missing output files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/tree_raw.nwk; Input files updated by another job: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/subsampled_alignment.fasta\n",
      "\n",
      "\n",
      "        augur tree             --alignment /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/subsampled_alignment.fasta             --tree-builder-args '-ninit 10 -n 4'             --output /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/tree_raw.nwk             --nthreads 4 2>&1 | tee /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/logs/tree_north-america.txt\n",
      "        \n",
      "\n",
      "[Tue Jun  9 01:47:48 2020]\n",
      "Job 6: Estimating censored KDE frequencies for tips\n",
      "Reason: Missing output files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/tip-frequencies.json; Input files updated by another job: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/data/metadata.tsv, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/tree.nwk\n",
      "\n",
      "\n",
      "        augur frequencies             --method kde             --metadata /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/data/metadata.tsv             --tree /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/tree.nwk             --min-date 2020.0             --pivot-interval 1             --narrow-bandwidth 0.05             --proportion-wide 0.0             --output /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/tip-frequencies.json 2>&1 | tee /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/logs/tip_frequencies_global.txt\n",
      "        \n",
      "\n",
      "[Tue Jun  9 01:47:48 2020]\n",
      "Job 23: \n",
      "        Inferring ancestral traits for ['region_exposure']\n",
      "          - increase uncertainty of reconstruction by 2.5 to partially account for sampling bias\n",
      "        \n",
      "Reason: Missing output files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/traits.json; Input files updated by another job: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/data/metadata.tsv, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/tree.nwk\n",
      "\n",
      "\n",
      "        augur traits             --tree /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/tree.nwk             --metadata /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/data/metadata.tsv             --output /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/traits.json             --columns region_exposure             --confidence             --sampling-bias-correction 2.5 2>&1 | tee /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/logs/traits_global.txt\n",
      "        \n",
      "\n",
      "[Tue Jun  9 01:47:48 2020]\n",
      "Job 17: \n",
      "        Reconstructing ancestral sequences and mutations\n",
      "          - inferring ambiguous mutations\n",
      "        \n",
      "Reason: Missing output files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/nt_muts.json; Input files updated by another job: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/subsampled_alignment.fasta, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/tree.nwk\n",
      "\n",
      "\n",
      "        augur ancestral             --tree /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/tree.nwk             --alignment /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/subsampled_alignment.fasta             --output-node-data /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/nt_muts.json             --inference joint             --infer-ambiguous 2>&1 | tee /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/logs/ancestral_global.txt\n",
      "        \n",
      "\n",
      "[Tue Jun  9 01:47:48 2020]\n",
      "Job 21: Adding internal clade labels\n",
      "Reason: Missing output files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/pangolin.json; Input files updated by another job: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/tree.nwk\n",
      "\n",
      "\n",
      "        /home/idies/workspace/Storage/sramakr4/persistent/miniconda3/envs/nextstrain/bin/python scripts/add_pangolin_lineages_local.py             --tree /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/tree.nwk             --clade /home/idies/workspace/covid19/jhu_sequences/submissions/clades/pangolin/latest/lineage_report_nextstrain.csv             --output /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/pangolin.json\n",
      "        \n",
      "\n",
      "[Tue Jun  9 01:47:48 2020]\n",
      "Job 18: Translating amino acid sequences\n",
      "Reason: Missing output files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/aa_muts.json; Input files updated by another job: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/tree.nwk, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/nt_muts.json\n",
      "\n",
      "\n",
      "        augur translate             --tree /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/tree.nwk             --ancestral-sequences /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/nt_muts.json             --reference-sequence config/reference.gb             --output-node-data /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/aa_muts.json 2>&1 | tee /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/logs/translate_global.txt\n",
      "        \n",
      "\n",
      "[Tue Jun  9 01:47:48 2020]\n",
      "Job 15: \n",
      "        Refining tree\n",
      "          - estimate timetree\n",
      "          - use skyline coalescent timescale\n",
      "          - estimate marginal node dates\n",
      "        \n",
      "Reason: Missing output files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/branch_lengths.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/tree.nwk; Input files updated by another job: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/tree_raw.nwk, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/subsampled_alignment.fasta, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/metadata_adjusted.tsv\n",
      "\n",
      "\n",
      "        augur refine             --tree /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/tree_raw.nwk             --alignment /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/subsampled_alignment.fasta             --metadata /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/metadata_adjusted.tsv             --output-tree /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/tree.nwk             --output-node-data /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/branch_lengths.json             --root Wuhan-Hu-1/2019 Wuhan/WH01/2019             --timetree             --clock-rate 0.0008             --clock-std-dev 0.0004             --coalescent skyline             --date-inference marginal             --divergence-unit mutations             --date-confidence             --no-covariance             --clock-filter-iqd 4 2>&1 | tee /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/logs/refine_north-america.txt\n",
      "        \n",
      "\n",
      "[Tue Jun  9 01:47:48 2020]\n",
      "Job 20: Adding internal clade labels\n",
      "Reason: Missing output files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/clades.json; Input files updated by another job: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/aa_muts.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/tree.nwk, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/nt_muts.json\n",
      "\n",
      "\n",
      "        augur clades --tree /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/tree.nwk             --mutations /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/nt_muts.json /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/aa_muts.json             --clades config/clades.tsv             --output-node-data /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/clades.json 2>&1 | tee /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/logs/clades_global.txt\n",
      "        \n",
      "\n",
      "[Tue Jun  9 01:47:48 2020]\n",
      "Job 36: Adding internal clade labels\n",
      "Reason: Missing output files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/temp_legacy_clades.json; Input files updated by another job: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/aa_muts.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/tree.nwk, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/nt_muts.json\n",
      "\n",
      "\n",
      "        augur clades --tree /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/tree.nwk             --mutations /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/nt_muts.json /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/aa_muts.json             --clades config/legacy_clades.tsv             --output-node-data /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/temp_legacy_clades.json 2>&1 | tee /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/logs/legacy_clades_global.txt\n",
      "        \n",
      "\n",
      "[Tue Jun  9 01:47:48 2020]\n",
      "Job 8: Estimating censored KDE frequencies for tips\n",
      "Reason: Missing output files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/tip-frequencies.json; Input files updated by another job: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/metadata_adjusted.tsv, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/tree.nwk\n",
      "\n",
      "\n",
      "        augur frequencies             --method kde             --metadata /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/metadata_adjusted.tsv             --tree /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/tree.nwk             --min-date 2020.0             --pivot-interval 1             --narrow-bandwidth 0.05             --proportion-wide 0.0             --output /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/tip-frequencies.json 2>&1 | tee /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/logs/tip_frequencies_north-america.txt\n",
      "        \n",
      "\n",
      "[Tue Jun  9 01:47:48 2020]\n",
      "Job 27: \n",
      "        Reconstructing ancestral sequences and mutations\n",
      "          - inferring ambiguous mutations\n",
      "        \n",
      "Reason: Missing output files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/nt_muts.json; Input files updated by another job: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/subsampled_alignment.fasta, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/tree.nwk\n",
      "\n",
      "\n",
      "        augur ancestral             --tree /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/tree.nwk             --alignment /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/subsampled_alignment.fasta             --output-node-data /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/nt_muts.json             --inference joint             --infer-ambiguous 2>&1 | tee /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/logs/ancestral_north-america.txt\n",
      "        \n",
      "\n",
      "[Tue Jun  9 01:47:48 2020]\n",
      "Job 31: Adding internal clade labels\n",
      "Reason: Missing output files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/pangolin.json; Input files updated by another job: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/tree.nwk\n",
      "\n",
      "\n",
      "        /home/idies/workspace/Storage/sramakr4/persistent/miniconda3/envs/nextstrain/bin/python scripts/add_pangolin_lineages_local.py             --tree /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/tree.nwk             --clade /home/idies/workspace/covid19/jhu_sequences/submissions/clades/pangolin/latest/lineage_report_nextstrain.csv             --output /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/pangolin.json\n",
      "        \n",
      "\n",
      "[Tue Jun  9 01:47:48 2020]\n",
      "Job 33: \n",
      "        Inferring ancestral traits for ['division_exposure']\n",
      "          - increase uncertainty of reconstruction by 2.5 to partially account for sampling bias\n",
      "        \n",
      "Reason: Missing output files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/traits.json; Input files updated by another job: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/metadata_adjusted.tsv, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/tree.nwk\n",
      "\n",
      "\n",
      "        augur traits             --tree /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/tree.nwk             --metadata /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/metadata_adjusted.tsv             --output /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/traits.json             --columns division_exposure             --confidence             --sampling-bias-correction 2.5 2>&1 | tee /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/logs/traits_north-america.txt\n",
      "        \n",
      "\n",
      "[Tue Jun  9 01:47:48 2020]\n",
      "rule rename_legacy_clades:\n",
      "    input: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/temp_legacy_clades.json\n",
      "    output: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/legacy_clades.json\n",
      "    jobid: 19\n",
      "    reason: Missing output files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/legacy_clades.json; Input files updated by another job: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/temp_legacy_clades.json\n",
      "    wildcards: build_name=global\n",
      "\n",
      "[Tue Jun  9 01:47:48 2020]\n",
      "Job 28: Translating amino acid sequences\n",
      "Reason: Missing output files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/aa_muts.json; Input files updated by another job: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/nt_muts.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/tree.nwk\n",
      "\n",
      "\n",
      "        augur translate             --tree /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/tree.nwk             --ancestral-sequences /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/nt_muts.json             --reference-sequence config/reference.gb             --output-node-data /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/aa_muts.json 2>&1 | tee /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/logs/translate_north-america.txt\n",
      "        \n",
      "\n",
      "[Tue Jun  9 01:47:48 2020]\n",
      "Job 40: Adding internal clade labels\n",
      "Reason: Missing output files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/temp_legacy_clades.json; Input files updated by another job: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/nt_muts.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/aa_muts.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/tree.nwk\n",
      "\n",
      "\n",
      "        augur clades --tree /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/tree.nwk             --mutations /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/nt_muts.json /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/aa_muts.json             --clades config/legacy_clades.tsv             --output-node-data /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/temp_legacy_clades.json 2>&1 | tee /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/logs/legacy_clades_north-america.txt\n",
      "        \n",
      "\n",
      "[Tue Jun  9 01:47:48 2020]\n",
      "Job 30: Adding internal clade labels\n",
      "Reason: Missing output files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/clades.json; Input files updated by another job: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/nt_muts.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/aa_muts.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/tree.nwk\n",
      "\n",
      "\n",
      "        augur clades --tree /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/tree.nwk             --mutations /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/nt_muts.json /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/aa_muts.json             --clades config/clades.tsv             --output-node-data /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/clades.json 2>&1 | tee /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/logs/clades_north-america.txt\n",
      "        \n",
      "\n",
      "[Tue Jun  9 01:47:48 2020]\n",
      "Job 9: Exporting data files for for auspice\n",
      "Reason: Missing output files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/ncov_with_accessions.json; Input files updated by another job: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/pangolin.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/legacy_clades.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/data/metadata.tsv, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/recency.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/clades.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/tree.nwk, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/aa_muts.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/branch_lengths.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/colors.tsv, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/traits.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/nt_muts.json\n",
      "\n",
      "\n",
      "        augur export v2             --tree /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/tree.nwk             --metadata /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/data/metadata.tsv             --node-data /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/branch_lengths.json /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/nt_muts.json /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/aa_muts.json /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/legacy_clades.json /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/clades.json /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/pangolin.json /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/recency.json /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/traits.json             --auspice-config profiles/jhu_release/global_auspice_config.json             --colors /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/colors.tsv             --lat-longs config/lat_longs.tsv             --title 'Genomic epidemiology of novel coronavirus - Global subsampling'             --description config/description.md             --output /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/ncov_with_accessions.json 2>&1 | tee /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/logs/export_global.txt\n",
      "        \n",
      "\n",
      "[Tue Jun  9 01:47:48 2020]\n",
      "Job 5: Adjusting main auspice JSON to take into account travel history\n",
      "Reason: Missing output files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/ncov_with_accessions_and_travel_branches.json; Input files updated by another job: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/ncov_with_accessions.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/colors.tsv\n",
      "\n",
      "\n",
      "        /home/idies/workspace/Storage/sramakr4/persistent/miniconda3/envs/nextstrain/bin/python ./scripts/modify-tree-according-to-exposure.py             --input /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/ncov_with_accessions.json             --colors /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/colors.tsv             --lat-longs config/lat_longs.tsv             --sampling region             --exposure region_exposure             --output /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/ncov_with_accessions_and_travel_branches.json 2>&1 | tee /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/logs/incorporate_travel_history_global.txt\n",
      "        \n",
      "\n",
      "[Tue Jun  9 01:47:48 2020]\n",
      "rule rename_legacy_clades:\n",
      "    input: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/temp_legacy_clades.json\n",
      "    output: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/legacy_clades.json\n",
      "    jobid: 29\n",
      "    reason: Missing output files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/legacy_clades.json; Input files updated by another job: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/temp_legacy_clades.json\n",
      "    wildcards: build_name=north-america\n",
      "\n",
      "[Tue Jun  9 01:47:48 2020]\n",
      "Job 1: Remove extraneous colorings for main build and move frequencies\n",
      "Reason: Missing output files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/auspice/ncov_global_tip-frequencies.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/auspice/ncov_global.json; Input files updated by another job: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/ncov_with_accessions_and_travel_branches.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/tip-frequencies.json\n",
      "\n",
      "\n",
      "        /home/idies/workspace/Storage/sramakr4/persistent/miniconda3/envs/nextstrain/bin/python scripts/fix-colorings.py             --input /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/ncov_with_accessions_and_travel_branches.json             --output /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/auspice/ncov_global.json 2>&1 | tee /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/logs/fix_colorings_global.txt &&\n",
      "        cp /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/tip-frequencies.json /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/auspice/ncov_global_tip-frequencies.json\n",
      "        \n",
      "\n",
      "[Tue Jun  9 01:47:48 2020]\n",
      "Job 13: Exporting data files for for auspice\n",
      "Reason: Missing output files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/ncov_with_accessions.json; Input files updated by another job: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/branch_lengths.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/aa_muts.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/colors.tsv, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/legacy_clades.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/nt_muts.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/clades.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/metadata_adjusted.tsv, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/recency.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/traits.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/pangolin.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/tree.nwk\n",
      "\n",
      "\n",
      "        augur export v2             --tree /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/tree.nwk             --metadata /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/metadata_adjusted.tsv             --node-data /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/branch_lengths.json /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/nt_muts.json /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/aa_muts.json /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/legacy_clades.json /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/clades.json /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/pangolin.json /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/recency.json /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/traits.json             --auspice-config profiles/jhu_release/north-america_auspice_config.json             --colors /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/colors.tsv             --lat-longs config/lat_longs.tsv             --title 'Genomic epidemiology of novel coronavirus - North America-focused subsampling'             --description config/description.md             --output /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/ncov_with_accessions.json 2>&1 | tee /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/logs/export_north-america.txt\n",
      "        \n",
      "\n",
      "[Tue Jun  9 01:47:48 2020]\n",
      "Job 3: Copying dated Auspice JSON\n",
      "Reason: Missing output files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/auspice/ncov_global_2020-06-09.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/auspice/ncov_global_2020-06-09_tip-frequencies.json; Input files updated by another job: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/tip-frequencies.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/auspice/ncov_global.json\n",
      "\n",
      "\n",
      "        cp /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/auspice/ncov_global.json /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/auspice/ncov_global_2020-06-09.json\n",
      "        cp /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/global/tip-frequencies.json /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/auspice/ncov_global_2020-06-09_tip-frequencies.json\n",
      "        \n",
      "\n",
      "[Tue Jun  9 01:47:48 2020]\n",
      "Job 7: Adjusting main auspice JSON to take into account travel history\n",
      "Reason: Missing output files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/ncov_with_accessions_and_travel_branches.json; Input files updated by another job: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/ncov_with_accessions.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/colors.tsv\n",
      "\n",
      "\n",
      "        /home/idies/workspace/Storage/sramakr4/persistent/miniconda3/envs/nextstrain/bin/python ./scripts/modify-tree-according-to-exposure.py             --input /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/ncov_with_accessions.json             --colors /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/colors.tsv             --lat-longs config/lat_longs.tsv             --sampling division             --exposure division_exposure             --output /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/ncov_with_accessions_and_travel_branches.json 2>&1 | tee /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/logs/incorporate_travel_history_north-america.txt\n",
      "        \n",
      "\n",
      "[Tue Jun  9 01:47:48 2020]\n",
      "Job 2: Remove extraneous colorings for main build and move frequencies\n",
      "Reason: Missing output files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/auspice/ncov_north-america.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/auspice/ncov_north-america_tip-frequencies.json; Input files updated by another job: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/ncov_with_accessions_and_travel_branches.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/tip-frequencies.json\n",
      "\n",
      "\n",
      "        /home/idies/workspace/Storage/sramakr4/persistent/miniconda3/envs/nextstrain/bin/python scripts/fix-colorings.py             --input /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/ncov_with_accessions_and_travel_branches.json             --output /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/auspice/ncov_north-america.json 2>&1 | tee /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/logs/fix_colorings_north-america.txt &&\n",
      "        cp /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/tip-frequencies.json /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/auspice/ncov_north-america_tip-frequencies.json\n",
      "        \n",
      "\n",
      "[Tue Jun  9 01:47:48 2020]\n",
      "Job 4: Copying dated Auspice JSON\n",
      "Reason: Missing output files: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/auspice/ncov_north-america_2020-06-09.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/auspice/ncov_north-america_2020-06-09_tip-frequencies.json; Input files updated by another job: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/auspice/ncov_north-america.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/tip-frequencies.json\n",
      "\n",
      "\n",
      "        cp /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/auspice/ncov_north-america.json /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/auspice/ncov_north-america_2020-06-09.json\n",
      "        cp /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/results/north-america/tip-frequencies.json /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/auspice/ncov_north-america_2020-06-09_tip-frequencies.json\n",
      "        \n",
      "\n",
      "[Tue Jun  9 01:47:48 2020]\n",
      "localrule all:\n",
      "    input: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/auspice/ncov_global.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/auspice/ncov_north-america.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/auspice/ncov_global_tip-frequencies.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/auspice/ncov_north-america_tip-frequencies.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/auspice/ncov_global_2020-06-09.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/auspice/ncov_north-america_2020-06-09.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/auspice/ncov_global_2020-06-09_tip-frequencies.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/auspice/ncov_north-america_2020-06-09_tip-frequencies.json\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/auspice/ncov_north-america_tip-frequencies.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/auspice/ncov_global_2020-06-09_tip-frequencies.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/auspice/ncov_north-america.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/auspice/ncov_global_2020-06-09.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/auspice/ncov_north-america_2020-06-09.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/auspice/ncov_global_tip-frequencies.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/auspice/ncov_global.json, /home/idies/workspace/Temporary/sramakr4/scratch/nextstrain_runs/alpha/ncov/output/auspice/ncov_north-america_2020-06-09_tip-frequencies.json\n",
      "\n",
      "Job counts:\n",
      "\tcount\tjobs\n",
      "\t1\tadjust_metadata_regions\n",
      "\t1\taggregate_alignments\n",
      "\t1\tall\n",
      "\t2\tancestral\n",
      "\t2\tclades\n",
      "\t2\tcolors\n",
      "\t2\tcombine_samples\n",
      "\t2\tdated_json\n",
      "\t1\tdiagnostic\n",
      "\t1\tdownload\n",
      "\t2\texport\n",
      "\t1\tfilter\n",
      "\t2\tfinalize\n",
      "\t2\tincorporate_travel_history\n",
      "\t2\tlegacy_clades\n",
      "\t1\tmask\n",
      "\t2\tpangolin\n",
      "\t1\tpartition_sequences\n",
      "\t1\tproximity_score\n",
      "\t2\trecency\n",
      "\t1\trefilter\n",
      "\t2\trefine\n",
      "\t2\trename_legacy_clades\n",
      "\t3\tsubsample\n",
      "\t2\ttip_frequencies\n",
      "\t2\ttraits\n",
      "\t2\ttranslate\n",
      "\t2\ttree\n",
      "\t47\n",
      "This was a dry-run (flag -n). The order of jobs does not reflect the order of execution.\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#run_cmd(\"snakemake -j 4 --unlock --forceall\")\n",
    "#run_cmd(\"snakemake -j 4 --forceall --snakefile Snakefile_Regions\")\n",
    "#run_cmd(\"snakemake -j 12 --forceall --snakefile Snakefile_main\")\n",
    "#run_cmd(\"snakemake --unlock\")\n",
    "#run_cmd(\"snakemake -j 14 --profile profiles/jhu_beta --snakefile Snakefile_main --dry-run\")\n",
    "run_cmd(\"snakemake -j 16 --profile profiles/jhu_release --snakefile Snakefile_main\")\n",
    "\n",
    "#run_cmd(\"snakemake -j 12 --snakefile Snakefile_main --dry-run\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.7 (py37)",
   "language": "python",
   "name": "py37"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
